presto-main/src/main/java/com/facebook/presto/execution/CreateTableTask.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.metadata.QualifiedTableName;
+import com.facebook.presto.metadata.TableHandle;
+import com.facebook.presto.metadata.TableMetadata;
+import com.facebook.presto.spi.ColumnMetadata;
+import com.facebook.presto.spi.ConnectorTableMetadata;
+import com.facebook.presto.sql.analyzer.SemanticException;
+import com.facebook.presto.sql.tree.CreateTable;
+import com.facebook.presto.sql.tree.TableElement;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeSignature;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.metadata.MetadataUtil.createQualifiedTableName;
+import static com.facebook.presto.sql.analyzer.SemanticErrorCode.TABLE_ALREADY_EXISTS;
+import static com.facebook.presto.sql.analyzer.SemanticErrorCode.MISSING_COLUMN_DEFINITION;
+
+public class CreateTableTask
+        implements DataDefinitionTask<CreateTable>
+{
+    @Override
+    public String getName()
+    {
+        return "CREATE TABLE";
+    }
+
+    @Override
+    public void execute(CreateTable statement, Session session, Metadata metadata, QueryStateMachine stateMachine)
+    {
+        QualifiedTableName tableName = createQualifiedTableName(session, statement.getTable());
+        Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
+        if (tableHandle.isPresent()) {
+            throw new SemanticException(TABLE_ALREADY_EXISTS, statement, "Table '%s' already exists", tableName);
+        }
+
+        if (statement.getTableElement().isEmpty()) {
+            throw new SemanticException(MISSING_COLUMN_DEFINITION, statement, "Table '%s' does not have column definition", tableName);
+        }
+
+        List<ColumnMetadata> createTableColumnList = new ArrayList<ColumnMetadata>();
+        int ordinalPosition = 0;
+        Iterator<TableElement> itr = statement.getTableElement().iterator();
+        while (itr.hasNext()) {
+            TableElement tblElement = itr.next();
+            Type columnType = metadata.getType(TypeSignature.parseTypeSignature(tblElement.getType().toLowerCase()));
```

Comment:
The `toLowerCase()` here is not necessary since `TypeSignature` handles this.

This needs handling for an unknown type and the `UNKNOWN` type. Check out `visitCast` in `ExpressionAnalyzer`.